### PR TITLE
Support create table as select

### LINKED ISSUE #6101: [Feature] Support create table as select...
**Is your feature request related to a problem? Please describe.**
Support create table as select...，This statement is handy for building ADS tables

**Describe the solution you'd like**
Get select field type to create table, It also presets some default configurations

### PR DESCRIPTION
**Pull request by stalary**
## Proposed changes

Support create table as select...，This statement is handy for building ADS tables

The decimal type is not currently supported

## Types of changes

What types of changes does your code introduce to Doris?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Code refactor (Modify the code structure, format the code, etc...)

Fix https://github.com/apache/incubator-doris/issues/6101

### CODE REVIEW

**Comment by stalary** [comment]
wait support string

**Comment by yangzhg** [comment]
> wait support string

String type has been merged

**Comment by stalary** [comment]
> > wait support string
> 
> String type has been merged

Ok, I will support it as soon as possible

File: fe/fe-core/src/main/java/org/apache/doris/catalog/Catalog.java [hunk 1]
```diff
@@ -3096,6 +3104,48 @@ public void createTableLike(CreateTableLikeStmt stmt) throws DdlException {
         }
     }
 
+    public void createTableAsSelect(CreateTableAsSelectStmt stmt) throws DdlException {
+        try {
+            QueryStmt tmpStmt = stmt.getQueryStmt().clone();
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/catalog/Catalog.java:3109]: Why not put this logic in `analyze()` of `CreateTableAsSelectStmt`?

File: fe/fe-core/src/main/java/org/apache/doris/catalog/Catalog.java [hunk 2]
```diff
@@ -3096,6 +3104,48 @@ public void createTableLike(CreateTableLikeStmt stmt) throws DdlException {
...
+            if (columnNames != null && columnNames.size() != tmpStmt.getColLabels().size()) {
+                ErrorReport.report(ErrorCode.ERR_COL_NUMBER_NOT_MATCH);
+            } else {
+                for (int i = 0; i < tmpStmt.getColLabels().size(); ++i) {
+                    String name;
+                    Expr expr = tmpStmt.getResultExprs().get(i);
+                    if (columnNames != null) {
+                        name = columnNames.get(i);
+                    } else {
+                        name = expr.toColumnLabel().replaceAll("'", "");
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/catalog/Catalog.java:3124]: How about decimal type

Comment (stalary) [reply fe/fe-core/src/main/java/org/apache/doris/catalog/Catalog.java]: Handle by reporting errors

File: fe/fe-core/src/main/java/org/apache/doris/catalog/Catalog.java (hunk 1 above)
Comment (stalary) [reply fe/fe-core/src/main/java/org/apache/doris/catalog/Catalog.java]: Ok, I'll modify it

**Review by morningman — COMMENTED** [review]
LGTM
Please resolve the conflict

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**